lib/SILOptimizer/Mandatory/TFPartition.cpp

```diff
@@ -3617,6 +3628,17 @@ insertTensorComputationStartEndTerminate(ArrayRef<SILValue> resultValues)
   };
 }
 
+/// Returns true when this function must be entirely lowered to a TF graph
+/// function, with no host-side logic remaining (i.e., no sends/recvs, and no
+/// start/stop tensor computation on the host side). In other words, this
+/// function uses the tensorflow calling convention.
+///
+/// The only way to call/use such a function is from a TF graph node (e.g. by
+/// referencing the function in a function-typed op attribute).
+static bool isAcceleratorOnly(const SILFunction &hostFn) {
+  // TODO: replace the impl with a proper function-level attribute.
+  return hostFn.getName().contains("__magic_tf_only__");
```

Comment:
Should this be put into the function classification thing at the top of the file, to keep the stringly typed stuff all in one place?

Reply:
Alternatively, should this start listening to @convention(TensorFlow)?  (Fine to do that as a follow-on patch if this is better).

Reply:
Which function classification logic are you referring to? (I only saw some instruction classification code at the top of the file.) 

This stringly-typed func impl is super hacky currently, and I left a comment to reimplement it. Note all call-sites are shielded from the stringly-type by calling this method, so it should be fairly easy to remove this hack later. 

As also suggested by @rxwei, the TODO comment involves re-implementing this method via listening to @convention(TensorFlow) -- I'll need to understand more about @convention(TensorFlow) vs the new @TensorFlowGraph first. :)

Reply:
In other words, `@TensorFlowGraph` is purely an AST-level sugar to work around the fact that the `@convention(...)` attribute is not a decl attribute and can only apply to types, whereas `@convention(tensorflow)` is carried all the way to SIL and IRGen, so you should check that :)